N/nativefiledialog_extended/build_tarballs.jl

```diff
@@ -0,0 +1,50 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "nativefiledialog_extended"
+version = v"1.1.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/btzy/nativefiledialog-extended/", "5786fabceeaee4d892f3c7a16b243796244cdddc", "nativefiledialog-extended-1.1.1")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd nativefiledialog-extended-1.1.1/
+if [[ "${target}" == *-mingw* ]]
+then
+    cmake -B build -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="-Wall -Wextra -pedantic" -DCMAKE_CXX_FLAGS="-Wall -Wextra -pedantic" -DBUILD_SHARED_LIBS=ON .
```

Comment:
Writing the `cmake` invocation like this, four times, makes it kinda difficult to understand what's going on and what the differences are. I would suggest to instead focus on the differences in the `if/elif/else` and then use that to adjust the arguments for a single `cmake build`.

Here, it seems dropping the `-Werror` would be one thing. Then perhaps `-DCMAKE_CXX_STANDARD=11` can just always be passed? That just leaves `-DNFD_PORTAL=ON/OFF` and `-DNFD_APPEND_EXTENSION=ON/OFF` as differences. So could invoke `cmake` like this (note the use of a multi-line command which further helps to see what is going on, and also when in the future this needs to be tweaked, the diff is much easier to read.
```
# set defaults
NFD_PORTAL=OFF
NFD_APPEND_EXTENSION=OFF
if [[ "${target}" == *-linux-* ]] ; then
  NFD_PORTAL=ON
elif [[ "${target}" == *-freebsd-* ]] ; then
  NFD_APPEND_EXTENSION=ON
fi

cmake -B build \
    -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_C_FLAGS="-Wall -Wextra -pedantic" \
    -DCMAKE_CXX_FLAGS="-Wall -Wextra -pedantic" \
    -DBUILD_SHARED_LIBS=ON \
    -DCMAKE_CXX_STANDARD=11 \
    -DNFD_PORTAL=${NFD_PORTAL} \
    -DNFD_APPEND_EXTENSION=${NFD_APPEND_EXTENSION} \
    .
```

Reply:
Changed. Still, I took the flags from the [`cmake.yml`](https://github.com/btzy/nativefiledialog-extended/blob/master/.github/workflows/cmake.yml) action. Do you know what version of `cmake` do we use; as in the actions they use the option `-G 'MSYS Makefiles'` which threw errors in the `buildkite`. 😅

Reply:
I don't think the `cmake` version is relevant here, but rather that we don't support targeting MSYS (which includes a POSIX emulation layer), but rather only MinGW. You could try using `-G 'MingGW Makefiles'` ? (And upstream perhaps also should use that, given that their cmake.yml only references MingGW, not MSYS.

Reply:
I'll let it out like this right now. As I'm more worried to be able to user to test it, as I only have Linux `x86_64`.